clang/include/clang/Sema/Sema.h

```diff
@@ -2245,8 +2249,12 @@ class Sema final : public SemaBase {
                             StringLiteral *SegmentName);
 
   /// Called on well-formed \#pragma init_seg().
+  /// If a function name is provided (", func-name"), it will be passed
+  /// via Func and FuncLoc.
   void ActOnPragmaMSInitSeg(SourceLocation PragmaLocation,
-                            StringLiteral *SegmentName);
+                            StringLiteral *SegmentName,
+                            IdentifierInfo *Func = nullptr,
```

Comment:
Just do an `IdentifierLoc` for these instead of 2 fields, thats what that is for!  Probably above as well.